crates/driver/src/util/bytes.rs

```diff
@@ -0,0 +1,30 @@
+/// A thin wrapper around a vector of bytes. Provides hex debug formatting.
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub struct Bytes<T>(pub T);
```

Comment:
I don't like storing these in structs to avoid having a manual Debug implementation. Now using the fields as intended is harder because you need `.0`, `.into()`, etc or some trait that exists for the raw bytes isn't implemented on the wrapper. I'm not extremely against it. For small structs like Uid and AppData a manual impl is better. For large structs I can see this have more use. An alternative is using a crate that derives Debug and allows you to annotate individual fields with custom formatters like in serde.

Reply:
This is similar to your [`ethcontract::Bytes`](https://docs.rs/ethcontract/latest/ethcontract/tokens/struct.Bytes.html), actually:

> Distinguishes a list of u8 from bytes.

Binary data is not the same thing as a list of (small) numbers. `Vec<u8>` is a list of numbers.

I agree that having to write `.0` is annoying. I personally don't mind it that much and am OK with it.